libp2p/autotls/acme/utils.nim

```diff
@@ -0,0 +1,43 @@
+import base64, strutils, chronos/apps/http/httpclient, json
+import ../../errors
+import ../../transports/tls/certificate_ffi
+
+type ACMEError* = object of LPError
+
+proc base64UrlEncode*(data: seq[byte]): string =
+  ## Encodes data using base64url (RFC 4648 §5) — no padding, URL-safe
+  var encoded = base64.encode(data, safe = true)
+  encoded.removeSuffix("=")
+  encoded.removeSuffix("=")
+  return encoded
+
+proc getResponseBody*(
+    response: HttpClientResponseRef
+): Future[JsonNode] {.async: (raises: [ACMEError, CancelledError]).} =
+  try:
+    let responseBody = bytesToString(await response.getBodyBytes()).parseJson()
+    return responseBody
+  except CancelledError as exc:
+    raise newException(CancelledError, "Future cancelled", exc)
```

Comment:
`raise newException(CancelledError, "Future cancelled", exc)` 
we don't really need to add any more context to `CancelledError`. when they happen we just need to return propagating back the same exception (`raise exc`). if this operation is of point of interest we can add logs. but exception will not be handled so adding any infromation will not help (as exception is just discarded).